src/document/actor.js

```diff
@@ -159,15 +159,23 @@ export class WarhammerActor extends WarhammerDocumentMixin(Actor)
             for (let uuid of effectUuids)
             {
                 let effect = await fromUuid(uuid);
+                let message = game.messages.get(messageId);
                 effectData.push(effect.convertToApplied(message?.system?.test));
             }
-
-            await CONFIG.ActiveEffect.documentClass.create(effectData, {parent: this, message : messageId});
-
+            for (let data of effectData)
+            {
+                await CONFIG.ActiveEffect.documentClass.create(data, {parent: this, message : messageId});
+            }
```

Comment:
What is the reason for this?

Reply:
for me - its for debugging purpose - it doesnt change that much in terms of performance, but its easier to verify which effect is corrupted.

Reply:
Well, I can sympathize, but let's go back to one database call for this.

Reply:
done